controller/backing_image_manager_controller.go

```diff
@@ -579,7 +579,9 @@ func (c *BackingImageManagerController) deleteInvalidBackingImages(bim *longhorn
 		err = errors.Wrapf(err, "failed to do cleanup for invalid backing images")
 	}()
 
-	for biName := range bim.Status.BackingImageFileMap {
+	for biName, biFileInfo := range bim.Status.BackingImageFileMap {
```

Comment:
note: the source of trough is backing image CR, so just base on it to determine which images are invalid in BIM CR.